core/config.h

```diff
@@ -58,6 +74,46 @@ class FuzzerConfig {
     [[nodiscard]] uint64_t getMinUpdateTimeInMicroseconds() const {
         return minUpdateTimeInMicroseconds;
     }
+
+    /// @brief Cast the TOML node to a specific type of value (encapsulated in `std::optional`).
+    template <typename T>
+    static std::optional<T> castTOMLNode(const toml::v3::node_view<const toml::v3::node> &node) {
+        if constexpr (std::is_same_v<T, int>) {
+            if (node.is_integer()) {
+                return node.as_integer()->get();
```

Comment:
Does get return `std::nullopt` when the cast fails? Or does it throw an exception?

Reply:
I am looking at the `node.as_integer()->get()`. If `node` is not an integer what happens here?

Reply:
If `node` is not an integer, it won't pass the `if (node.is_integer())` check where [`is_integer` (supported in TOML++)](https://marzer.github.io/tomlplusplus/classtoml_1_1node.html#ac5390ac64f018947edc75802a53505a5) would "return true if this node is an integer value."

Reply:
Oh. Sorry that I probably misunderstood your question. A [`std::nullptr`](https://marzer.github.io/tomlplusplus/classtoml_1_1table.html#a5b86281620ba34efabf7e3c4bef9675b) would be returned if `node` is not a node representation of an integer.

Reply:
Ah. Yes. I will then make the function return a `std::nullptr` if the node is not a representation of the expected underlying type (e.g., integer) and, correspondingly, move the `get()` to the caller's definition since it won't make sense if the `get` function is called on a potential `std::nullptr`.